src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -832,11 +812,12 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     <ItemGroup>
       <ResolvedCompileFileDefinitions Remove="@(_PublishConflictPackageFiles)" Condition="'%(_PublishConflictPackageFiles.ConflictItemType)' == 'Reference'" />
-      <NativeCopyLocalItems Remove="@(_PublishConflictPackageFiles)" Condition="'%(_PublishConflictPackageFiles.ConflictItemType)' != 'Reference'" />
-      <ResourceCopyLocalItems Remove="@(_PublishConflictPackageFiles)" Condition="'%(_PublishConflictPackageFiles.ConflictItemType)' != 'Reference'" />
-      <RuntimeCopyLocalItems Remove="@(_PublishConflictPackageFiles)" Condition="'%(_PublishConflictPackageFiles.ConflictItemType)' != 'Reference'" />
       <RuntimeTargetsCopyLocalItems Remove="@(_PublishConflictPackageFiles)" Condition="'%(_PublishConflictPackageFiles.ConflictItemType)' != 'Reference'" />
       <RuntimePackAsset Remove="@(_PublishConflictPackageFiles)" Condition="'%(_PublishConflictPackageFiles.ConflictItemType)' != 'Reference'" />
+
+      <_ResolvedNuGetFilesForPublish Include="@(NativeCopyLocalItems);@(ResourceCopyLocalItems);@(RuntimeCopyLocalItems)" />
+      <_ResolvedNuGetFilesForPublish Remove="@(_PublishConflictPackageFiles)" Condition="'%(_PublishConflictPackageFiles.ConflictItemType)' != 'Reference'" />
+      <_ResolvedNuGetFilesForPublish Remove="@(_ResolvedNuGetFilesForPublish)" Condition="'%(_ResolvedNuGetFilesForPublish.CopyToPublishDirectory)' == 'false'" />
```

Comment:
I wish there were some way to combine this with the `Include`.  Perhaps three includes, one for each item type (`NativeCopyLocalItems`, `ResourceCopyLocalItems`, and `RuntimeCopyLocalItems`) that is conditional as not being excluded from publish?  Not that users typically have large sets of packages, but I'd hate to dump them all in a set and then have to traverse the set to remove them again.

Reply:
I'm not sure using three separate lists would end up being more efficient, since they are going to be combined to pass to the task anyway.  Putting them in one list means we don't have to duplicate the Remove code, so I think it's preferable unless we're somewhat confident it could be a perf issue.

Reply:
Not three separate lists, three includes into the same list.

I.e.:

```xml
<_ResolvedNuGetFilesForPublish Include="@(NativeCopyLocalItems)" Condition="'%(NativeCopyLocalItems.CopyToPublishDirectory)' == 'false'" />
<_ResolvedNuGetFilesForPublish Include="@(ResourceCopyLocalItems)" Condition="'%(ResourceCopyLocalItems.CopyToPublishDirectory)' == 'false'" />
<_ResolvedNuGetFilesForPublish Include="@(RuntimeCopyLocalItems)" Condition="'%(RuntimeCopyLocalItems.CopyToPublishDirectory)' == 'false'" />
```